libcudacxx/test/libcudacxx/cuda/barrier/cp_async_bulk_tensor_generic.h

```diff
@@ -129,10 +130,10 @@ __constant__ fake_cutensormap global_fake_tensor_map;
  * 5. It writes the tile back to global memory
  * 6. It checks that all the values in global are properly modified.
  */
-template <int smem_len>
-__device__ void test(std::initializer_list<int> smem_coord,
-                     std::initializer_list<int> smem_dims,
-                     std::initializer_list<int> gmem_dims,
+template <size_t smem_len, size_t num_dims>
+__device__ void test(cuda::std::array<uint32_t, smem_len> smem_coord,
```

Comment:
```suggestion
__device__ void test(cuda::std::array<uint32_t, num_dims> smem_coord,
```

smem_len is the size of the shared memory in bytes. It should not be set by the number of dimensions of the shared memory coordinate.